hierarchy/entity.cpp

```diff
@@ -0,0 +1,68 @@
+#include "entity.h"
+#include <cstring>
+entity::entity()
+{
+
+}
+
+entity::entity(const entity & toAdd)
+{
+	memId.add(toAdd.memId);	
+}
+
+entity::~entity()
+{
+
+}
+
+bool entity::addId()
+{
+	char temp[ID];
+	using namespace std;
+	cout<<"Please enter a new User Identification number: ";
+	cin.get(temp, ID, '\n');
+	cin.ignore(100, '\n');
+
+	int length = strlen(temp);
+
+	while(length < 9 || length > 9)
```

Comment:
This should be in a function, we're going to be doing a lot of it.

Reply:
I added a function that returned a char * that basically took all this code and returned the correct ID. Does that work?